jbpm-console-ng-generic/jbpm-console-ng-generic-client/src/main/java/org/jbpm/console/ng/gc/client/experimental/workbench/configuration/WorkbenchConfigurationPopup.java

```diff
@@ -0,0 +1,338 @@
+            boolean refreshWorkbenchFlag) {
+        if (refreshWorkbenchFlag == false) {
+            return;
+        }
+        if (selectedLanguageItem == null || selectedLanguageItem.equals("")) {
+            showFieldEmptyWarning();
+        } else {
+            setCurrentLanguage(selectedLanguageItem);
+        }
+    }
+
+    /**
+     * Multiple view mode function
+     **/
+
+    private List<Node<String, String>> getAllMultipleModes() {
+        List<Node<String, String>> allMultipleMode = new ArrayList<Node<String, String>>(
+                2);
+        allMultipleMode.add(new Node<String, String>(constants.Advanced(),
+                contextualView.ADVANCED_MODE));
+        allMultipleMode.add(new Node<String, String>(constants.Basic(),
+                contextualView.BASIC_MODE));
+        return allMultipleMode;
+
+    }
+
+    private void setMultipleModeItems(List<Node<String, String>> items) {
+        multipleModeItems.clear();
+        for (Node<String, String> item : items) {
+            multipleModeItems.addItem(item.getKey(), item.getValue());
+        }
+
+    }
+
+    private Node<String, String> getSelectedMultipleModeItem() {
+        final int selectedIndex = multipleModeItems.getSelectedIndex();
+        if (selectedIndex == -1) {
+            return new Node<String, String>("", "");
+        }
+        final String text = multipleModeItems.getItemText(selectedIndex);
+        final String value = multipleModeItems.getValue(selectedIndex);
+        return new Node<String, String>(text, value);
+    }
+
+    private void switchMode(boolean refreshPerspectiveFlag,
+            boolean refreshWorkbenchFlag) {
+        if (refreshPerspectiveFlag && !refreshWorkbenchFlag) {
+            String modeName = contextualView.getModeName();
+            if (modeName.equals(BASIC_MODE)) {
+                contextualView.setModeName(ADVANCED_MODE);
+            } else {
+                contextualView.setModeName(BASIC_MODE);
+            }
+            refreshPerspective();
+        }
+    }
+
+    private void refreshPerspective() {
+        final PerspectiveActivity currentPerspective = perspectiveManager
+                .getCurrentPerspective();
+        perspectiveManager
+                .removePerspectiveStates(new org.uberfire.mvp.Command() {
+                    @Override
+                    public void execute() {
+                        if (currentPerspective != null) {
+                            final PlaceRequest pr = new ForcedPlaceRequest(
+                                    currentPerspective.getIdentifier(),
+                                    currentPerspective.getPlace()
+                                            .getParameters());
+                            placeManager.goTo(pr);
+                        }
+                    }
+                });
+    }
+
+    private void showFieldEmptyWarning() {
+        ErrorPopup.showMessage(constants.PleaseSetAName());
+    }
+
+    class Node<K, V> {
```

Comment:
yes, but Pair can not use in jbpm-console-ng, because of some rules in here .
if this code move to uberfire, i will change it